### PR TITLE
Expose current routes context pathname as useRoutePathname()

### PR DESCRIPTION
**Pull request by damikun**
There should be a way to get current routes context pathname.

Equal is done as useOutlet() for outlet and  useParams() for params

This change adds useRoutesPath() and Returns the pathname element under current router context...

**Edited:** by @timdorr recomendation will name useRoutePathname()

### CODE REVIEW

File: packages/react-router/index.tsx [hunk 1]
```diff
@@ -463,6 +463,15 @@ export function useParams(): Params {
   return React.useContext(RouteContext).params;
 }
 
+ /**
+ * Returns the pathname element under current router context.
+ *
+ * @see https://reactrouter.com/api/useRouterPath
+ */
+export function useRoutesPath(): string {
```
Comment (timdorr) [inline packages/react-router/index.tsx:471]:
```suggestion
export function useRoutePathname(): string {
```

File: packages/react-router/index.tsx [hunk 2]
```diff
@@ -463,6 +463,15 @@ export function useParams(): Params {
   return React.useContext(RouteContext).params;
 }
 
+ /**
+ * Returns the pathname element under current router context.
+ *
+ * @see https://reactrouter.com/api/useRouterPath
```
Comment (timdorr) [inline packages/react-router/index.tsx:469]:
```suggestion
 * @see https://reactrouter.com/api/useRoutePathname
```

**Comment by timdorr** [comment]
The naming should be consistent and `useRoutePathname` is probably a more descriptive name.

**Comment by damikun** [comment]
Yes sounds better..  All build is ok just final cross-package exports fails do you know what is that ?

I think it is some compare if all exports are in booth packages react-router and react-router-dom but not sure about this check on end on all test what is idea behind it..

```
 FAIL  packages/react-router-native/__tests__/exports-test.js
```

File: packages/react-router/index.tsx (hunk 2 above)
Comment (damikun) [reply packages/react-router/index.tsx:469]: Yes that sound better..

**Comment by damikun** [comment]
I fixed missing reexports for useRoutePathname and build is ok now...

**Review by Hummel37 — APPROVED** [review]

**Comment by mjackson** [comment]
Thanks for the PR @damikun. What is your use case? We are not likely to merge new API unless we understand where our current API falls short.

**Comment by damikun** [comment]
Hi @mjackson its been while since I create it :)  but I think it was explained in this post: https://github.com/remix-run/react-router/issues/7735

**Comment by Hummel37** [comment]
@mjackson @damikun 
Can the pathname not get it's own context? Currently the useNavigate hook for example needs the pathname too and rerenders every time you change anything. That would be really nice and save much rerenderers.

**Comment by ryanflorence** [comment]
```js
useResolvedPath(".").pathname
```

**Pull request closed without merging**